docs/how_to/extending_page_title.rst

```diff
@@ -291,14 +293,7 @@ low-level API to edit the toolbar according to your needs::
                 # Nothing to do
                 return
 
-            # check global permissions if CMS_PERMISSION is active
-            if get_cms_setting('PERMISSION'):
-                has_global_current_page_change_permission = has_page_change_permission(self.request)
-            else:
-                has_global_current_page_change_permission = False
-                # check if user has page edit permission
-            can_change = self.request.current_page and self.request.current_page.has_change_permission(self.request)
-            if has_global_current_page_change_permission or can_change:
+            if user_can_change_page(user=self.request.user, page=self.page):
```

Comment:
Yes, in the example code we set `self.page = get_page_draft(self.request.current_page)`, this is because the request can come from a live page. Totally missed that in your diff :/.

I don't think it's necessary to set `self.page` but always use the draft page, so you can set a local variable `page` but make sure to use `get_page_draft`